### ISSUE TITLE
Slow execution of phpcbf

### ISSUE THREAD

**Issue by shivanandwp** [soderlind/vscode-phpcbf#2]
which phpcs
/usr/local/bin/phpcs

which phpcbf
/usr/local/bin/phpcbf

phpcbf -i
The installed coding standards are PEAR, Zend, PSR2, MySource, Squiz, PSR1, WordPress-VIP, WordPress, WordPress-Extra, WordPress-Docs and WordPress-Core

Settings:
"phpcbf.standard": "WordPress",

**Comment by shivanandwp** [comment]
So it takes several minutes to update the fixed file. I'm still trying to do console.log and see which part of the extension is taking all this while. Or may be msvsc takes minutes to detect file changes and reload the file?

**Comment by shivanandwp** [comment]
So I edited
```
let exec = cp.spawn(this.executablePath, this.getArgs(fileName));
console.log("spawned at:" + Date.now());
```
and
```exec.on("exit", code => {
console.log("Exit status:" + code);
console.log("Finished at:" + Date.now());```

Output:
spawned at:1515061443180
Finished at:1515061503327

This seems like the time difference in milliseconds was 60147 which is about 60 seconds. So it works. Does it take as long on your side?

**Comment by shivanandwp** [comment]
Please also see:
https://github.com/Glavin001/atom-beautify/issues/893
https://github.com/nodejs/node/issues/7652
https://github.com/Glavin001/atom-beautify/pull/1059

**Comment by shivanandwp** [comment]
I just closed the stdin and now it does it in seconds. Hope this is the correct way.

"Time: 135ms; Memory: 8Mb"

```
fs.unlink(fileName, function(err) {});
        autoFixing = false;
      });
    });

	exec.stdin.end();
	  
    exec.stdout.on("data", buffer => {
      console.log(buffer.toString());
    });
```

**Comment by soderlind** [comment]
excellent, thank you :) I'll add, test and push a new version later today (after work).

**Comment by soderlind** [comment]
@shivanandwp I believe I should close stdin on close, I'm a newb at writing extensions so I might be wrong:
```javascript
exec.on("close", code => {
	exec.stdin.end();
});
```

**Comment by soderlind** [comment]
Forget my last comment, closing stdin just after the promise, as you suggested, makes the execution much faster.

**Comment by shivanandwp** [comment]
I'm totally newbie but I can mess around. I think it still waits for stdin before it hits close. I tried your code but it still hit a minute.

I then placed the end directly after the exec declaration. This does it in `Time: 132ms; Memory: 8Mb`

```
let exec = cp.spawn(this.executablePath, this.getArgs(fileName));
exec.stdin.end();
```

**Comment by soderlind** [comment]
I've been reading up on child processes https://nodejs.org/api/child_process.html. Tried to use execFile, but it failed. I'll move `exec.stdin.end();` just after `cp.spawn`, test a bit more and push the code to the marketplace.

**Comment by soderlind** [comment]
Fixed in 0.0.3
